project/api.py

```diff
@@ -0,0 +1,204 @@
+from datetime import datetime, timedelta
+import flask
+from flask import request, jsonify
+from werkzeug.exceptions import HTTPException
+import pandas as pd
+
+
+app = flask.Flask(__name__)
+
+
+def to_milliseconds(dt):
+    """Convert datetime object to timestamp in millisecond format"""
+    return int(datetime.timestamp(dt)*1000.)
+
+
+def prepare_request(time_range_start, time_range_end):
+    """Convert datetime time range to  timestamp in millisecond format"""
+    datetime_start = datetime.strptime(time_range_start, '%Y-%m-%d %H:%M:%S')
+    datetime_end = datetime.strptime(time_range_end, '%Y-%m-%d %H:%M:%S')
+    datetime_start_code = to_milliseconds(datetime_start)
+    datetime_end_code = to_milliseconds(datetime_end)
+    return datetime_start_code, datetime_end_code
+
+
+@app.route('/', methods=['GET'])
+def home():
+    return "<h1>Junior Data Scientist/Data Engineer Test Task</h1>"
+
+
+@app.errorhandler(Exception)
+def handle_error(e):
+    """Generic exception handler"""
+    code = 500
+    if isinstance(e, HTTPException):
+        code = e.code
+    return jsonify(error=str(e)), code
+
+
+@app.errorhandler(404)
+def page_not_found(e):
+    """Handle cases with empty query parameter"""
+    code = 404
+    return jsonify(error=str(e), code=code)
+
+
+@app.errorhandler(400)
+def handle_bad_request(e):
+    """Handle cases with wrong query parameter"""
+    code = 400
+    return jsonify(error=str(e), code=code), code
+
+
+@app.route('/api/v1/queries/avg_elapsed_time', methods=['GET'])
+def avg_elapsed_time():
+    """Returns an average elapsed time for all queries on certain date"""
+    query_parameters = request.args
+    date = query_parameters.get('date')
+    if not date:
+        return page_not_found('No date provided')
+
+    data = pd.read_csv('queries.tsv', sep='\t')
```

Comment:
What if you call your application from another directory? Will it still find the `queries.tsv`?